src/models/CustomAction.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-08-19.
+ */
+
+// TS2019-DONE
+
+// Create
+export interface ICreateCustomActionParams {
+  sourceKey: string;
+  name: string;
+  urlTemplate: string;
+  description: string;
+  sharing: CustomActionSharing;
+}
+export interface ICreateCustomActionResponse extends ICustomAction {}
+
+// Delete
+export interface IDeleteCustomActionParams {
+  sourceKey: string;
+  id: number;
+}
+
+// Update
+export interface IUpdateCustomActionParams {
+  sourceKey: string;
+  id: number;
+  name?: string;
+  urlTemplate?: string;
+  description?: string;
+  sharing?: CustomActionSharing;
+}
+export interface IUpdateCustomActionResponse extends ICustomAction {}
+
+// Get
+export interface IGetCustomActionsParams {
+  sourceKey: string;
+}
+export type IGetCustomActionsResponse = ICustomAction[];
+
+// Entities
+export interface ICustomAction {
```

Comment:
you should make it extend: IPersistedItem
most likely we want in the API response also createdAt and updatedAt